File: Python/my_first_class_node.py
#!/usr/bin/env python3

import rclpy # biblioteca padrão do ros para o python
from rclpy.node import Node

class MyNode(Node):
    def __init__(self):
        super().__init__("py_class_test")
        self.get_logger().info("Hello World") # printa algo no terminal

def main(args = None):
    rclpy.init(args = args) # inicializa a comunicação com o ros
    node = MyNode()
    rclpy.spin(node) # Faz com que o nó, continue "rodando", até pressionar "ctrl c". #1 nome do nó
    rclpy.shutdown() # esta normalmente é a linha final do nó

if __name__ == '__main__':
    main()

File: C++/my_first_class_node.cpp
#include "rclcpp/rclcpp.hpp"

class MyNode : public rclcpp::Node{
    public:
        MyNode() : Node("cpp_class_test"){
            RCLCPP_INFO(this->get_logger(), "Hello World");
        }
    private:
};

int main(int argc, char **argv){

    rclcpp::init(argc, argv);
    auto node = std::make_shared<MyNode>()
    rclcpp::spin(node)
    rclcpp::shutdown()

    return 0;
}
